bodhi/tests/server/test_metadata.py

```diff
@@ -342,48 +348,72 @@ def _test_extended_metadata(self, has_alias):
-             'TurboGears-1.0.2.2-2.fc17.src.rpm'))
-        self.assertEqual(pkg.version, '1.0.2.2')
-        self.assertFalse(pkg.reboot_suggested)
-        self.assertEqual(pkg.arch, 'src')
-        self.assertEqual(pkg.filename, 'TurboGears-1.0.2.2-2.fc17.src.rpm')
+        updateinfos = self._verify_updateinfos(self.repodata)
+
+        for updateinfo in updateinfos:
+            # Read an verify the updateinfo.xml.gz
+            uinfo = createrepo_c.UpdateInfo(updateinfo)
+            notice = self.get_notice(uinfo, 'mutt-1.5.14-1.fc13')
+            self.assertIsNone(notice)
+
+            self.assertEqual(len(uinfo.updates), 1)
+            notice = uinfo.updates[0]
+
+            self.assertIsNotNone(notice)
+            self.assertEqual(notice.title, update.title)
+            self.assertEqual(notice.release, update.release.long_name)
+            self.assertEqual(notice.status, update.status.value)
+            if update.date_modified:
+                self.assertEqual(notice.updated_date, update.date_modified)
+            self.assertEqual(notice.fromstr, config.get('bodhi_email'))
+            self.assertEqual(notice.rights, config.get('updateinfo_rights'))
+            self.assertEqual(notice.description, update.notes)
+            self.assertEqual(notice.id, update.alias)
+            self.assertEqual(notice.severity, 'Moderate')
+            bug = notice.references[0]
+            self.assertEqual(bug.href, update.bugs[0].url)
+            self.assertEqual(bug.id, '12345')
+            self.assertEqual(bug.type, 'bugzilla')
+            cve = notice.references[1]
+            self.assertEqual(cve.type, 'cve')
+            self.assertEqual(cve.href, update.cves[0].url)
+            self.assertEqual(cve.id, update.cves[0].cve_id)
+
+            col = notice.collections[0]
+            self.assertEqual(col.name, update.release.long_name)
+            self.assertEqual(col.shortname, update.release.name)
+
+            pkg = col.packages[0]
+            self.assertEqual(pkg.epoch, '0')
+            self.assertEqual(pkg.name, 'TurboGears')
+            self.assertEqual(
+                pkg.src,
+                ('https://download.fedoraproject.org/pub/fedora/linux/updates/testing/17/SRPMS/T/'
+                 'TurboGears-1.0.2.2-2.fc17.src.rpm'))
+            self.assertEqual(pkg.version, '1.0.2.2')
+            self.assertFalse(pkg.reboot_suggested)
+            self.assertEqual(pkg.arch, 'src')
+            self.assertEqual(pkg.filename, 'TurboGears-1.0.2.2-2.fc17.src.rpm')
+
+    # Make sure we reach 100% coverage, even though cr.ZCK_COMPRESSION is only defined
+    # when createrepo_c supports zchunk
+    @mock.patch('bodhi.server.metadata.cr')
+    def test_zchunk_metadata(self, mock_cr):
+        mock_cr.ZCK_COMPRESSION = 99
+        assert(bodhi_metadata.cr.ZCK_COMPRESSION == 99)
+        try:
+            bodhi_metadata.insert_in_repo(99, self.tempcompdir, 'garbage', 'zck', '/dev/null')
+        except TypeError:
```

Comment:
What would raise this ```TypeError```, and the other two below?

Reply:
Without this, I get 
E           TypeError: write() argument must be str, not MagicMock
at the line: repomd_file.write(repomd.xml_dump()) in insert_in_repo() in bodhi/server/metadata.py

Reply:
It seems to me that this probably happens because we need to do more mocking to get it to work - i.e., we need to make our mock work such that it hands out a ```str``` to that ```write()``` call.